integration/v3_watch_test.go

```diff
@@ -479,15 +479,19 @@ func TestV3WatchCurrentPutOverlap(t *testing.T) {
 	// last mod_revision that will be observed
 	nrRevisions := 32
 	// first revision already allocated as empty revision
+	var wg sync.WaitGroup
 	for i := 1; i < nrRevisions; i++ {
+		wg.Add(1)
 		go func() {
+			defer wg.Done()
 			kvc := toGRPC(clus.RandClient()).KV
 			req := &pb.PutRequest{Key: []byte("foo"), Value: []byte("bar")}
 			if _, err := kvc.Put(context.TODO(), req); err != nil {
 				t.Errorf("couldn't put key (%v)", err)
 			}
 		}()
 	}
+	wg.Wait()
```

Comment:
I think this way the for loop below will only be executed after all the put, which seems not serving the purpose of test. We should move the wg.wait to after the for loop so that the overlap put can be watched. Thanks!